gobblin-config-management/src/main/java/gobblin/config/client/ConfigClient.java

```diff
@@ -0,0 +1,122 @@
+package gobblin.config.client;
+
+import java.net.URI;
+import java.util.Collection;
+import java.util.Map;
+
+import com.typesafe.config.Config;
+
+import gobblin.config.configstore.ConfigStore;
+
+/**
+ * ConfigClient is the consumer facing class for retrieving configuration against all configuration stores.
+ * @author mitu
+ *
+ */
+public class ConfigClient {
+
+  /**
+   * RAISE_ERROR 
+   * @author mitu
+   *
+   */
+  public static enum VERSION_STABILITY_POLICY{
+    RAISE_ERROR,
+    CACHE_CONFIG_IN_MEMORY
+  }
+  
+  /**
+   * @param policy - the policy control the behavior when this configuration client try to connect to configuration store.
+   * if policy is RAISE_ERROR and this client try to connect to any {@gobblin.config.configstore.ConfigStore} which is not
+   * {@gobblin.config.configstore.ConfigStoreWithStableVersion}, the RuntimeException will be thrown
+   * 
+   * if policy is CACHE_CONFIG_IN_MEMORY and this client try to connect to any {@gobblin.config.configstore.ConfigStore} which is not
+   * {@gobblin.config.configstore.ConfigStoreWithStableVersion}, the consumer need to cache the retrieved configuration in memory,
+   * otherwise, the same query for the same URI against same configuration store may result different result.
+   * @return - the Configuration Client object
+   */
+  public static ConfigClient createConfigClientWithStableVersion( VERSION_STABILITY_POLICY policy ){
+    return null;
+  }
+  
+  /**
+   * 
+   * @return the {@ConfigClient} with VERSION_STABILITY_POLICY.CACHE_CONFIG_IN_MEMORY
+   */
+  public static ConfigClient createDefaultConfigClient( ){
```

Comment:
I would prefer that the default is RAISE_ERROR ( CACHE_CONFIG_IN_MEMORY can easily become a memory leak)